cores/esp8266/core_esp8266_waveform.cpp

```diff
@@ -256,23 +256,24 @@ static ICACHE_RAM_ATTR void timer1Interrupt() {
         // Check for toggles
         int32_t cyclesToGo = wave->nextServiceCycle - now;
         if (cyclesToGo < 0) {
+          cyclesToGo = -((-cyclesToGo) % (wave->nextTimeHighCycles + wave->nextTimeLowCycles));
```

Comment:
Logically, I get what you're trying to here.  But it seems like this is a no-op unless we have overshot by an entire waveform cycle, no?  If it's been less than a waveform cycle then the mod will be a no-op.   The mod operator is a relatively expensive one, might consider just ignoring the case for code speed.

Reply:
Correct, this should be a no-op most of the time. It's a safeguard because if we overshoot one or multiple wafeforms, the PWM will oscillate at 500KHz (1ms up + 1ms down) until the required number of cycles is reached. I don't know if it's even possible, or if 500KHz for a very short amount of time is an issue at all.
We could also add a `while (-cyclesToGo > wave->nextTimeHighCycles + wave->nextTimeLowCycles) { cyclesToGo += wave->nextTimeHighCycles + wave->nextTimeLowCycles)};` but that would also add code size to the precious IRAM.

I'm pretty sure this would never occur with PWM, only if someone sets a very short waveform. You can probably drop this for now.

Reply:
Could you comment it out and make a note in the code about why we're skipping, so if something pops up later we can see the reasoning behind the choice?

Reply:
I removed this line of code for nom but leaved the two variants in comments.